### PR TITLE
Prac 5 codę review request

### PR DESCRIPTION
**Pull request by Corey-Grainger**
@jeomin12 Please review my code from prac_05 for any errors or opportunities for improvement.  Specifically I think there might be a better approach to extracting the name from the input email in the emails program.

### CODE REVIEW

File: prac_05/state_names.py [hunk 1]
```diff
@@ -11,8 +11,10 @@
 
 state_code = input("Enter short state: ").upper()
 while state_code != "":
-    if state_code in CODE_TO_NAME:
+    try:
         print(state_code, "is", CODE_TO_NAME[state_code])
```
Comment (jeomin12) [inline prac_05/state_names.py:15]: You can use f-strings to format the output for easy readability.

File: prac_05/hex_colours.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+""" CP1404 Week 5 Practical
+ Hex Colours Program"""
+
+NAME_TO_HEX_CODE = {"Candy Apple Red": "#ff0800", "Banana Yellow": "#ffe135", "Banana Mania": "#fae7b5",
+                    "Apricot": "#fbceb1", "Asparagus": "#87a96b", "Bitter Lime": "#bfff00", "Black Olive": "#3b3c36",
+                    "Boysenberry": "#873260", "Carrot Orange": "#ed9121", "Deep Peach": "#ffcba4",
+                    "Eggplant": "#614051", "Granny Smith Apple": "#a8e4a0", "Mango Tango": "#ff8243"}
+
+print(NAME_TO_HEX_CODE)
+name = input("Enter colour name: ").title()
```
Comment (jeomin12) [inline prac_05/hex_colours.py:10]: Can use a better variable name which can be specific

File: prac_05/word_occurences.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+"""CP1404 Week 5 Prac
+Word Occurrences Program
+Estimated: 25 minutes
+Actual: 15 minutes"""
+
+string = input("Text: ").lower()
+words = string.split()
+words.sort()
+words_to_characters_in_word = {}
```
Comment (jeomin12) [inline prac_05/word_occurences.py:14]: Use a better variable name. words_to_characters_in_word is a confusing name.

File: prac_05/wimbledon.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
+"""CP1404 Week 05 Prac
+Wimbledon Program"""
+
+FILENAME = "wimbledon.csv"
+
+
+def main():
+    """Load a CSV file of wimbledon stats and print selected stats."""
+    wimbledon_records = load_file(FILENAME)
+    player_to_number_of_wins = extract_player_number_of_wins(wimbledon_records)
```
Comment (jeomin12) [inline prac_05/wimbledon.py:10]: player_to_number_of_wins variable could be used as champions

File: prac_05/wimbledon.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    print("These 12 countries have won Wimbledon: ")
+    print(", ".join(wimbledon_winning_countries))
+
+
+def load_file(filename):
+    """Load data from a CSV file."""
+    lines = []
+    with open(filename, "r", encoding="utf-8-sig") as in_file:
+        in_file.readline()
+        for line in in_file:
```
Comment (jeomin12) [inline prac_05/wimbledon.py:32]: You can use list comprehensions to make your code more concise.

File: prac_05/wimbledon.py [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+def display_wimbledon_winner_statistics(player_to_number_of_wins, wimbledon_winning_countries):
+    """Display wimbledon winner statistics."""
+    print("Wimbledon Champions: ")
+    longest_name_length = max((len(name) for name in player_to_number_of_wins))
+    print(longest_name_length)
+    for player, number_of_wins in player_to_number_of_wins.items():
+        print(f"{player:{longest_name_length}} - {number_of_wins}")
+    print()
+    print("These 12 countries have won Wimbledon: ")
```
Comment (jeomin12) [inline prac_05/wimbledon.py:23]: Consider avoiding numbers like "12" when printing the number of winning countries. Instead, you can calculate the count dynamically according to data

Comment (Corey-Grainger) [reply prac_05/wimbledon.py:23]: This is a good point. I should have derived this number. 👍

File: prac_05/wimbledon.py (hunk 4 above)
Comment (Corey-Grainger) [reply prac_05/wimbledon.py:10]: As is champions_to_number_of_wins instead?  Is there a way champions is better than player?

File: prac_05/word_occurences.py (hunk 3 above)
Comment (Corey-Grainger) [reply prac_05/word_occurences.py:14]: Good point.  I think word_to_character_count is probably a bit more concise and readable.

**Pull request merged**